web/app.js

```diff
@@ -1329,12 +1329,33 @@ const PDFViewerApplication = {
       pdfViewer.optionalContentConfigPromise.then(optionalContentConfig => {
         this.pdfLayerViewer.render({ optionalContentConfig, pdfDocument });
       });
+      if ("requestIdleCallback" in window) {
+        window.requestIdleCallback(() => {
+          this._collectTelemetry();
+        });
+      }
     });
 
     this._initializePageLabels(pdfDocument);
     this._initializeMetadata(pdfDocument);
   },
 
+  /**
+   * A place to fetch data for telemetry after one page is rendered and the
+   * viewer is idle.
+   */
+  async _collectTelemetry() {
+    if (!this.pdfDocument) {
+      return;
+    }
+    const markInfo = await this.pdfDocument.getMarkInfo();
```

Comment:
I'd suggest the following here, which is consistent with the various `this._initializeX` methods that exists below:
```suggestion
  async _collectTelemetry(pdfDocument) {
    const markInfo = await pdfDocument.getMarkInfo();
    
    if (pdfDocument !== this.pdfDocument) {
      return; // The document was closed while the MarkInfo data resolved.
    }
```